packages/circuit-ui/components/Popover/Popover.tsx

```diff
@@ -266,30 +269,39 @@ export const Popover = ({
     ) {
       return;
     }
-    onClose(event);
+    setOpen(false);
   });
 
-  if (!isOpen) {
-    return null;
-  }
-
   return (
     <Fragment>
-      <Overlay />
-      <PopoverWrapper
-        {...props}
-        ref={setPopperElement}
-        style={styles.popper}
-        {...attributes.popper}
-      >
-        {actions.map((action, index) =>
-          isDivider(action) ? (
-            <Hr css={dividerStyles} key={index} />
-          ) : (
-            <PopoverItem key={index} {...action} />
-          ),
-        )}
-      </PopoverWrapper>
+      <Component
+        ref={triggerRef}
+        id={triggerid}
+        aria-haspopup={true}
+        aria-controls={id}
+        onClick={() => setOpen((prev) => !prev)}
+      />
+      {isOpen && (
+        <Fragment>
+          <Overlay />
+          <PopoverWrapper
+            id={id}
+            {...props}
+            ref={setPopperElement}
+            style={styles.popper}
+            aria-labelledby={triggerid}
+            {...attributes.popper}
+          >
+            {actions.map((action, index) =>
+              isDivider(action) ? (
+                <Hr css={dividerStyles} key={index} />
```

Comment:
Do we have a test with a divider?

Reply:
👀 I don't think so. We could add it to the base `actions` so that it makes it into the snapshots